src/http_server.rs

```diff
@@ -1384,48 +1384,36 @@ mod tests {
             .post("/login")
             // panic if form is not deserializable...
             .form(&cred)
+            .expect_failure()
             .await;
-        // 303 here...
-        assert_eq!(res.status_code(), StatusCode::SEE_OTHER);
-        // get cookie
-        let res_headers = res.headers();
-        assert!(res_headers.contains_key("set-cookie"));
-        let cookie = match res_headers.get("set-cookie") {
-            Some(cookie) => cookie.clone(),
-            None => panic!(),
-        };
```

Comment:
The manual cookie handling to grab it, and pass it back up, is all built into Axum Test already. On line 1371 I add `client.do_save_cookies();` which enables this (it will also expire cookies if the server does that in a response).